ethcore/src/header.rs

```diff
@@ -186,11 +186,12 @@ impl Header {
 	/// Get the hash of the header excluding the seal
 	pub fn bare_hash(&self) -> H256 {
 		let mut hash = self.bare_hash.borrow_mut();
-		match &mut *hash {
+		match hash.deref_mut() {
```

Comment:
> deref coercions are applied repeatedly .... c++'s -> operator ... acts similarly

c++'s `->` operator does not apply dereferences repeatedly until a match is found.

i see that by doing this, rust allows things like `Box` to work much as `unique_ptr` does in C++, though the manner in which it manages the syntactic feature makes the overall semantics for dereferencing "more flexible" - and thus less explicit - than in C++. a particularly glaring fact given how anal retentive the language otherwise tends to be.